### PR TITLE
Use dynatrace-metric-utils

### PR DESCRIPTION
**Pull request by dyladan**

### CODE REVIEW

File: plugins/outputs/dynatrace/dynatrace.go [hunk 1]
```diff
@@ -101,163 +88,92 @@ func (d *Dynatrace) Description() string {
...
-				default:
-					fmt.Fprintf(&buf, "%s%s %v\n", metricID, tagb.String(), value)
-				}
+			name := tm.Name() + "." + field.Key
+			dm, err := dtMetric.NewMetric(
+				name,
+				dtMetric.WithPrefix(d.Prefix),
+				dtMetric.WithDimensions(
+					dimensions.MergeLists(
+						// dimensions.NewNormalizedDimensionList(e.opts.DefaultDimensions...),
```
Comment (pirgeo) [inline plugins/outputs/dynatrace/dynatrace.go:143]:
```suggestion
```

File: plugins/outputs/dynatrace/dynatrace.go [hunk 2]
```diff
@@ -101,163 +88,92 @@ func (d *Dynatrace) Description() string {
...
+			typeOpt := getTypeOption(metricType, field)
 
-				metricType := metric.Type()
-				for _, i := range d.AddCounterMetrics {
-					if metric.Name()+"."+metricKey == i {
-						metricType = telegraf.Counter
-					}
-				}
+			if typeOpt == nil {
+				// unsupported type
```
Comment (pirgeo) [inline plugins/outputs/dynatrace/dynatrace.go]: should we log something here?

File: plugins/outputs/dynatrace/dynatrace.go (hunk 1 above)
Comment (dyladan) [reply plugins/outputs/dynatrace/dynatrace.go:143]: I left this there as a reminder. We currently don't have the default dimensions functionality. I was going to add it in a separate PR

Comment (pirgeo) [reply plugins/outputs/dynatrace/dynatrace.go:143]: Oh, I see. Feel free to ignore this then!

**Review by pirgeo — APPROVED** [review]
LGTM

**Pull request closed without merging**